public/app/plugins/datasource/elasticsearch/datasource.test.ts

```diff
@@ -58,73 +72,68 @@ const createTimeRange = (from: DateTime, to: DateTime): TimeRange => ({
   },
 });
 
-interface Args {
-  data?: any;
+interface TestContext {
+  data?: Data;
   from?: string;
-  jsonData?: any;
+  jsonData?: Partial<ElasticsearchOptions>;
   database?: string;
-  mockImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+  fetchImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+}
+
+interface Data {
+  [key: string]: undefined | string | string[] | number | Data | Data[];
 }
 
 function getTestContext({
-  data = {},
+  data = { responses: [] },
   from = 'now-5m',
-  jsonData = {},
-  database = '[asd-]YYYY.MM.DD',
-  mockImplementation = undefined,
-}: Args = {}) {
-  jest.clearAllMocks();
-
+  jsonData,
+  database = '[test-]YYYY.MM.DD',
+  fetchImplementation = undefined,
+}: TestContext = {}) {
   const defaultMock = (options: BackendSrvRequest) => of(createFetchResponse(data));
 
   const fetchMock = jest.spyOn(backendSrv, 'fetch');
-  fetchMock.mockImplementation(mockImplementation ?? defaultMock);
+  fetchMock.mockImplementation(fetchImplementation ?? defaultMock);
 
-  const templateSrv: any = {
-    replace: jest.fn((text?: string) => {
+  const timeSrv = {
+    time: { from, to: 'now' },
+    timeRange: () => ({
+      from: dateMath.parse(timeSrv.time.from, false),
+      to: dateMath.parse(timeSrv.time.to, true),
+    }),
+    setTime: (time: RawTimeRange) => {
+      timeSrv.time = time;
+    },
+  } as TimeSrv;
+
+  const settings: Partial<DataSourceInstanceSettings<ElasticsearchOptions>> = { url: ELASTICSEARCH_MOCK_URL };
+  if (jsonData) {
+    settings.jsonData = jsonData as ElasticsearchOptions;
+  }
+
+  const templateSrv = {
+    replace: (text?: string) => {
       if (text?.startsWith('$')) {
         return `resolvedVariable`;
       } else {
         return text;
       }
-    }),
-    getAdhocFilters: jest.fn(() => []),
-  };
+    },
+    getAdhocFilters: () => [],
+  } as unknown as TemplateSrv;
 
-  const timeSrv: any = {
-    time: { from, to: 'now' },
-  };
-
-  timeSrv.timeRange = jest.fn(() => {
-    return {
-      from: dateMath.parse(timeSrv.time.from, false),
-      to: dateMath.parse(timeSrv.time.to, true),
-    };
-  });
-
-  timeSrv.setTime = jest.fn((time) => {
-    timeSrv.time = time;
-  });
-
-  const instanceSettings: DataSourceInstanceSettings<ElasticsearchOptions> = {
-    id: 1,
-    meta: {} as DataSourcePluginMeta,
-    name: 'test-elastic',
-    type: 'type',
-    uid: 'uid',
-    access: 'proxy',
-    url: ELASTICSEARCH_MOCK_URL,
-    database,
-    jsonData,
-    readOnly: false,
-  };
-
-  const ds = new ElasticDatasource(instanceSettings, templateSrv);
+  // @ts-expect-error
```

Comment:
I don't like that we kinda suppress the error here. 
The issue is, that you are using the "wrong" `TemplateSrc`. If you use this import instead, everything will work:
`import { TemplateSrv } from 'app/features/templating/template_srv';`

Reply:
Honestly, I'm not sure why that works because usually there's no simple way to mock `TemplateSrv` without TS saying that it doesn't overlap enough.